x-pack/plugins/security_solution/public/common/components/exceptions/viewer/exceptions_viewer_header.tsx

```diff
@@ -0,0 +1,204 @@
+      filter: { filter, showDetectionsList, showEndpointList, tags },
+      pagination: {},
+    });
+  }, [filter, tags, showDetectionsList, showEndpointList, onFilterChange]);
+
+  const onAddExceptionDropdownClick = useCallback(
+    (): void => setAddExceptionMenuOpen(!isAddExceptionMenuOpen),
+    [setAddExceptionMenuOpen, isAddExceptionMenuOpen]
+  );
+
+  const handleDetectionsListClick = useCallback((): void => {
+    setShowDetectionsList(!showDetectionsList);
+    setShowEndpointList(false);
+  }, [showDetectionsList, setShowDetectionsList, setShowEndpointList]);
+
+  const handleEndpointListClick = useCallback((): void => {
+    setShowEndpointList(!showEndpointList);
+    setShowDetectionsList(false);
+  }, [showEndpointList, setShowEndpointList, setShowDetectionsList]);
+
+  const handleOnSearch = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>): void => {
+      const searchValue = event.target.value;
+      const tagsRegex = /(tags:[^\s]*)/i;
+      const tagsMatch = searchValue.match(tagsRegex);
+      const foundTags: string = tagsMatch != null ? tagsMatch[0].split(':')[1] : '';
+      const filterString = tagsMatch != null ? searchValue.replace(tagsRegex, '') : searchValue;
+
+      if (foundTags.length > 0) {
+        setTags(foundTags.split(','));
+      }
+
+      setFilter(filterString.trim());
+    },
+    [setTags, setFilter]
+  );
+
+  const onAddException = useCallback(
+    (type: ExceptionListType): void => {
+      onAddExceptionClick(type);
+      setAddExceptionMenuOpen(false);
+    },
+    [onAddExceptionClick, setAddExceptionMenuOpen]
+  );
+
+  const addExceptionButtonOptions = useMemo(
+    (): EuiContextMenuPanelDescriptor[] => [
+      {
+        id: 0,
+        items: [
+          {
+            name: i18n.ADD_TO_ENDPOINT_LIST,
+            onClick: () => onAddException(ExceptionListType.ENDPOINT),
+            'data-test-subj': 'addEndpointExceptionBtn',
+          },
+          {
+            name: i18n.ADD_TO_DETECTIONS_LIST,
+            onClick: () => onAddException(ExceptionListType.DETECTION_ENGINE),
+            'data-test-subj': 'addDetectionsExceptionBtn',
+          },
+        ],
+      },
+    ],
+    [onAddException]
+  );
+
+  return (
+    <EuiFlexGroup alignItems="center">
+      <EuiFlexItem grow={true}>
+        <EuiFieldSearch
+          data-test-subj="exceptionsHeaderSearch"
+          aria-label={i18n.SEARCH_DEFAULT}
+          placeholder={i18n.SEARCH_DEFAULT}
+          onChange={handleOnSearch}
+          disabled={isInitLoading}
+          incremental={false}
+          fullWidth
+        />
+      </EuiFlexItem>
+
+      {supportedListTypes.length < 2 && (
+        <EuiFlexItem grow={false}>
+          <EuiButton
+            data-test-subj="exceptionsHeaderAddExceptionBtn"
+            onClick={() => onAddException(supportedListTypes[0])}
+            isDisabled={isInitLoading}
+            fill
+          >
+            {i18n.ADD_EXCEPTION_LABEL}
```

Comment:
Thanks for adding support for hiding irrelevant filters/actions -- tested it out and looks good! Not sure if we'll want specific copy here based on which list we'll be adding the exception to, but we can wait and see how we'll manage which rules have support for endpoint exception lists before addressing.

Reply:
Thanks for testing!